src/org/labkey/test/tests/component/GridPanelViewTest.java

```diff
@@ -0,0 +1,1389 @@
+import org.labkey.test.params.FieldDefinition;
+import org.labkey.test.params.experiment.SampleTypeDefinition;
+import org.labkey.test.util.APIUserHelper;
+import org.labkey.test.util.ApiPermissionsHelper;
+import org.labkey.test.util.DataRegionTable;
+import org.labkey.test.util.PermissionsHelper;
+import org.labkey.test.util.SampleTypeHelper;
+import org.labkey.test.util.TestDataGenerator;
+import org.labkey.test.util.exp.SampleTypeAPIHelper;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Category({Daily.class})
+public class GridPanelViewTest extends GridPanelBaseTest
+{
+
+    // Sample type used to validate views.
+    private static final String DEFAULT_VIEW_SAMPLE_TYPE = "View_SampleType";
+    private static final int DEFAULT_VIEW_SAMPLE_TYPE_SIZE = 30;
+    private static final String DEFAULT_VIEW_SAMPLE_PREFIX = "VIEW-";
+
+    // A sample type that will be used to validate the dialog. Will test adding 'hidden' fields etc...
+    private static final String VIEW_DIALOG_ST = "Test_View_Dialog_SampleType";
+    private static final int VIEW_DIALOG_ST_SIZE = 100;
+    private static final String VIEW_DIALOG_ST_PREFIX = "DLG-";
+
+    // Column names.
+    private static final String COL_NAME = "Name";
+    private static final String COL_STRING1 = "Str1";
+    private static final String COL_STRING2 = "Str2";
+    private static final String COL_INT = "Int";
+    private static final String COL_BOOL = "Bool";
+
+    private static final List<String> DEFAULT_COLUMNS = Arrays.asList(COL_NAME, COL_INT, COL_STRING1, COL_STRING2, COL_BOOL);
+
+    // Will keep track of state of the columns, that is are they filtered, sorted, or have no modifiers.
+    private static Map<String, Integer> defaultColumnState = new HashMap<>();
+    private static final int ICONS_NONE = 0;
+    private static final int ICONS_SORT = 1;
+    private static final int ICONS_FILTER = 2;
+
+    private static final String OTHER_USER = "other_user@grid.panel.test";
+
+    private static final List<String> stringSetMembers = Arrays.asList("A", "B", "C");
+    private static List<String> stringSets = new ArrayList<>();
+
+    private static final String EDITED_ALERT = "EDITED";
+    private static final String UPDATED_ALERT = "UPDATED";
+
+    // Keep track of any custom views that may have been created.
+    private static List<String> savedViewsForDefaultSampleType = new ArrayList<>();
+
+    // Using the core-components.view adds 'GridPanel - ' to the panel header. Need to take that into account .
+    private static final String PANEL_VIEW_NAME_PREFIX = "GridPanel - %s";
+
+    // Tests that need to be written:
+    // Validate "Save As..." from the grid save button.
+    // Validate views that are locked, or in some other way, cannot be updates in the manage views dialog.
+    // ???
```

Comment:
do you have this list somewhere to fill out this ???

Reply:
I've added a story to Samurai Puffin back log to track this (Grid Views -Selenium Test Part 2).
It includes these test and a couple of others.